src/main/java/com/minecraftabnormals/abnormals_core/core/events/CompatEvents.java

```diff
@@ -59,4 +74,25 @@ public static void onUpdateEntity(LivingEvent.LivingUpdateEvent event) {
 			if (entity.getPersistentData().getBoolean(POISON_TAG)) ((IAgeableEntity) entity).resetGrowthProgress();
 		}
 	}
+
+	@SubscribeEvent(priority = EventPriority.LOWEST)
+	public static void onNoteBlockPlay(NoteBlockEvent.Play event) {
+		if (SORTED_CUSTOM_NOTE_BLOCK_INSTRUMENTS  != null) {
+			World world = (World) event.getWorld();
+			if (!world.isClientSide()) {
+				BlockPos pos = event.getPos();
+				for (DataUtil.CustomNoteBlockInstrument instrument : SORTED_CUSTOM_NOTE_BLOCK_INSTRUMENTS) {
+					if (instrument.test(new ProxyBlockSource((ServerWorld) world, pos.relative(Direction.DOWN)))) {
```

Comment:
`ProxyBlockSource` doesn't change for every iteration, meaning it should be a local variable.

Reply:
done